### PR TITLE
Reboot

### PR DESCRIPTION
**Pull request by sarayourfriend**
So far everything works except downloading files. Not sure what I did wrong but currently the frontend doesn't link to the S3 URL (but admin does).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,11 +76,6 @@ migrate: .env
 	$(MANAGE) migrate
 
 
-.PHONY: types
```
Comment (AetherUnbound) [inline Makefile:79]: This is causing one of the checks to fail, should we just have the one check for pre-commit?

File: compose/local/django/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
-FROM python:3.8-slim
+FROM python:3.9-slim
```
Comment (AetherUnbound) [inline compose/local/django/Dockerfile:1]: :raised_hands:

File: pra_request_tracker/pra_request_tracker/templates/base.html [hunk 3]
```diff
@@ -68,25 +68,6 @@
             <li class="nav-item">
               <a class="nav-link" href="{% url 'record_requests:agencies' %}">Agencies</a>
             </li>
-            {% if request.user.is_authenticated %}
```
Comment (AetherUnbound) [inline pra_request_tracker/pra_request_tracker/templates/base.html:71]: Why was this removed?

File: production.yml [hunk 4]
```diff
@@ -35,7 +35,7 @@ services:
     ports:
       - "6667:5000"
     volumes:
-      - static:/app/pra_request_tracker/static
+      - static:/app/pra_request_tracker/pra_request_tracker/static
```
Comment (AetherUnbound) [inline production.yml]: Oh this was something I found out yesterday, since this the app is running `collectstatic` as part of `/entrypoint`, we actually want this to point to `/app/staticfiles` instead

**Review by AetherUnbound — COMMENTED** [review]
Thanks so much for taking this on! What's the issue you're seeing with links in non-admin pages?

File: production.yml (hunk 4 above)
Comment (AetherUnbound) [reply production.yml]: I also had to add this to the dockerfile too:
```
RUN mkdir -p /app/staticfiles && chown django /app/staticfiles
```

File: pra_request_tracker/pra_request_tracker/templates/base.html (hunk 3 above)
Comment (sarayourfriend) [reply pra_request_tracker/pra_request_tracker/templates/base.html:71]: We didn't really have anything for the front-end for logged in users, just the admin panel right?

File: Makefile (hunk 1 above)
Comment (sarayourfriend) [reply Makefile:79]: Oh! I meant to mention in the PR, I could not for the life of me get mypy to work correctly so I stripped it out. It was failing on all sorts of weird issues in the settings.py file. I'll remove it from the workflows.

Comment (AetherUnbound) [reply Makefile:79]: Totally fine with that lol, mypy is good but can be a real PITA sometimes and for this particular PR it's too many moving parts.

File: pra_request_tracker/pra_request_tracker/templates/base.html (hunk 3 above)
Comment (AetherUnbound) [reply pra_request_tracker/pra_request_tracker/templates/base.html:71]: Ahh, fair point!

**Comment by sarayourfriend** [comment]
> What's the issue you're seeing with links in non-admin pages?

In non-admin pages it isn't linking to the S3 url... For example I uploaded a file and it links to `http://localhost:8000/record-requests/1/file.png` instead of where the admin links `http://minio:9000/prr-tracker/qr-code.png?AWSAccessKeyId=minio&Signature=XuDrTPKBm%2BavlJf4DEKMEa%2F%2BtIA%3D&Expires=1626364604` (which doesn't work to download the file either but I'm assuming that's a minio thing...)

**Comment by sarayourfriend** [comment]
Okay I figured it out. Something about how the file was being serialized before was magically turning it into the URL, but now we have to ask for the url from the file specifically 🤷 Not a big deal.

Only thing is that locally it links to `http://minio:9000/...` which doesn't work unless you've set up your hosts file the same way docker's internal network is working. What do you think @AetherUnbound what's the best solution here?

File: pra_request_tracker/pra_request_tracker/templates/record_requests/recordrequest_detail.html [hunk 5]
```diff
@@ -28,7 +28,7 @@ <h1>{{ object.title }}</h1>
       {% for file in object.files %}
         <ul>
           <li>
-            <a href="{{ file.file }}">{{ file.title }}</a>
+            <a href="{{ file.file.url }}">{{ file.title }}</a>
```
Comment (AetherUnbound) [inline pra_request_tracker/pra_request_tracker/templates/record_requests/recordrequest_detail.html:31]: Ahhhhh

File: .env.example.prod [hunk 6]
```diff
@@ -18,13 +17,3 @@ AWS_S3_REGION_NAME=us-west-2
 AWS_S3_ENDPOINT_URL=https://sfo3.digitaloceanspaces.com
 
 REDIS_URL=redis://redis
-
-DJANGO_DEFAULT_FROM_EMAIL=techblocsea@protonmail.com
-DJANGO_SERVER_EMAIL=error@openoversight.tech-bloc-sec.dev
-
-EMAIL_HOST=mail
```
Comment (AetherUnbound) [inline .env.example.prod:25]: Does that mean we should remove the `protonmail` network from `production.yml` too?

**Review by AetherUnbound — COMMENTED** [review]
It seems like `static/project.css` and `static/project.js` are both empty files, is that intended?

As for the minio issue - UGH I'm so sorry, that's an issue we ran into with OO as well. I made a change in [this PR](https://github.com/SeattleDSA/OpenOversight/commit/467d1cc96887e68dc5e70a210639069764f7f922#diff-8ea1a2f8f85606363b9141c143067b028617fc0fa5e537a59df795076a9b804dR225) (hopefully that links directly to the file but it's in `utils.py`) that changed the URL to `localhost` if `minio` is in it. I don't love that solution 😕 Particularly here because we have less control over what the filename looks like so someone could just put `minio` in it and replicate the logic on production. So we could have more guardrails or figure out something different entirely, maybe using IP addresses locally instead of `localhost`?

**Comment by sarayourfriend** [comment]
> It seems like static/project.css and static/project.js are both empty files, is that intended?

Yes, I was just using them to test the static files were working. We don't really need anything in there so I guess we can remove them.

**Comment by sarayourfriend** [comment]
> So we could have more guardrails or figure out something different entirely, maybe using IP addresses locally instead of localhost?

I mean we could just request that development environments add `minio    127.0.0.1` to their hosts files?

**Comment by sarayourfriend** [comment]
Not sure why tests fail in CI they run fine locally 🤔

**Comment by AetherUnbound** [comment]
> I mean we could just request that development environments add `minio 127.0.0.1` to their hosts files?

Eh, I was trying to avoid that but since we're adding it in more projects that seems fine.

**Review by AetherUnbound — APPROVED** [review]
So trimmed down, I love it!

**Comment by AetherUnbound** [comment]
> Not sure why tests fail in CI they run fine locally 🤔

Maybe the `COPY pytest.ini /pytest.ini` needs to be `COPY pytest.ini /app/pytest.ini`?
